addon/mixins/oauth2-implicit-grant-callback-route-mixin.js

```diff
@@ -0,0 +1,102 @@
+import Ember from 'ember';
+
+const { inject: { service }, Mixin, testing, computed, getOwner } = Ember;
+
+/**
+  __This mixin is used in the callback route when using OAuth 2.0 Implicit
+  Grant authentication.__ It implements the
+  {{#chrossLink "OAuth2ImplicitGrantCallbackRouteMixin/activate:method"}}{{/crossLink}}
+  method that retrieves and processes authentication parameters, such as
+  `access_token`, from the hash parameters provided in the callback URL by
+  the authentication server. The parameters are then passed to the
+  {{#crossLink "OAuth2ImplicitGrantAuthenticator"}}{{/crossLink}}
+
+  @class OAuth2ImplicitGrantCallbackMixin
+  @module ember-simple-auth/mixins/ouath2-implicit-grant-callback-mixin
+  @extends Ember.Mixin
+  @public
+*/
+
+export default Mixin.create({
+  /**
+   The session service.
+
+   @property session
+   @readOnly
+   @type SessionService
+   @public
+   */
+  session: service('session'),
+
+  /**
+    The authenticator that should be used to authenticate the callback. This
+    must be a subclass of the
+    {{#crossLink "OAuth2ImplicitGrantAuthenticator"}}{{/crossLink}}
+    authenticator.
+
+    @property authenticator
+    @type String
+    @default null
+    @public
+  */
+  authenticator: null,
+
+  /**
+    Any error that potentially occurs during authentication will be stored in
+    this property.
+
+    @property error
+    @type String
+    @default null
+    @public
+  */
+  error: null,
+
+  /**
+    Passes the hash received with the redirection from the authentication
+    server to the
+    {{#crossLink "OAuth2ImplicitGrantAuthenticator"}}{{/crossLink}} and
+    authenticates the session with the authenticator.
+
+    @method activate
+    @public
+  */
+  activate() {
+    if (!testing && this.get('_isFastBoot')) {
```

Comment:
I think it was because I couldn't get the getOwner to return anything in testing mode, and _isFastBoot depends on getOwner. In retrospect there's probably a better way to handle that but I wasn't sure what it was.